### ISSUE TITLE
Android app shows ntfy icon at top even when there are no notifications

### ISSUE THREAD

**Issue by chrisco484** [binwiederhier/ntfy#1109]
:bulb: **Idea**
Currently on Android the ntfy icon is always present if the app is installed/running - even when there are no active notifications.

When I expand all notifications I can normally slide left to acknowledge most notifications and they disappear but the nfty one can not be removed.

This is rather annoying because I normally only like to see icons in the top bar when there are active/unacknowledged notification that require my attention.

Most other apps like FB, Whatsapp, LinkedIn, GMail etc., appear to implement this 'convention' of 'if you don't see an app's icon in the top bar then there's no notifications associated with the app' but unfortunately ntfy does not appear to follow this convention and its icon is present in the top bar even if there are no active notifications - unless there's a configuration setting that I'm missing?

:computer: **Target components**

Android app

**Comment by wunter8** [comment]
In recent Android versions, you can hide the persistent notification at a system settings level. What Android version are you using?

**Comment by chrisco484** [comment]
> In recent Android versions, you can hide the persistent notification at a system settings level. What Android version are you using?

Android 13

**Comment by wunter8** [comment]
You should be able to just swipe the notification away. Also, if you long press the notification, and click on the gear/settings icon in the corner, it should bring you to the notification settings page where you can hide the notification entirely.

**Comment by eerison** [comment]
I'm with the same problem, the icon just disappear when I turn off the notifications 🥲

**Comment by chrisco484** [comment]
> I'm with the same problem, the icon just disappear when I turn off the notifications 🥲

Yeah, it's annoying heh! If there isn't anything to notify me about I don't want to see that icon at all. If every app that could potentially have notifications showed a permanent icon in the top section of my phone's screen it would be cluttered with icons and I would not be able to tell which ones actually had real notifications and which ones didn't - defeating the purpose of that section of the Android screen.

**Comment by binwiederhier** [comment]
You are likely talking about the foreground service. This is a technical limitation on Android. The foreground service does not run if you're using firebase, which ntfy.sh uses, but your self hosted server cannot use. 

You can hide it like this: https://docs.ntfy.sh/subscribe/phone/#instant-delivery

**Comment by chrisco484** [comment]
Ah, ok, so it's the self hosting aspect that causes the problem. I think I understand it now.

Is there a paid service whereby our self hosted service can send an event to the ntfy.sh service so that it can produce the notifications in Android? So then, presumably, Android sees the notification coming from ntfy.sh and so it behaves like a normal notification icon that can be removed without needing to do the workaround of disabling instant notifications?

**Comment by binwiederhier** [comment]
You can read all about it here: https://blog.ntfy.sh/2023/12/06/138-lines-of-code/#android-app

If you are technical, you can sign up for Firebase (see https://docs.ntfy.sh/develop/#firebase-setup), run your own ntfy server, and compile your own Android app with the Firebase service file embedded. Then you won't have the foreground service and notifications will come from Firebase/Google directly.

Happy to answer more questions, but this ticket is answered by https://docs.ntfy.sh/subscribe/phone/#instant-delivery -- You CAN disable the foreground service.

**Comment by mattokar** [comment]
Hi. 
Where exactly should `Subscription Service` option to disable be? I can't find it in app settings nor in app's notification settings page.
Thank you

**Comment by kinslayer1982** [comment]
@binwiederhier 
> You can hide it like this: https://docs.ntfy.sh/subscribe/phone/#instant-delivery

Sorry, but the link is wrong. This is absolutely NOT how it looks like when following the instruction and there is nothing even similar to "subscription service".

**Comment by binwiederhier** [comment]
> Android does not allow you to dismiss this notification, unless you turn off the notification channel in the settings. To do so, long-press on the foreground notification (screenshot above) and navigate to the settings. Then toggle the "Subscription Service" off

It may look slightly different on your phone, but the instructions still work on my OnePlus phone

**Comment by kinslayer1982** [comment]
I'd say this is more than just slightly different.

![Image](https://github.com/user-attachments/assets/72121019-0084-4b7d-bc55-807bc59e8170)
